src/main/java/edu/kis/vh/stacks/stack.java

```diff
@@ -1,34 +1,48 @@
 package edu.kis.vh.stacks;
 
+import edu.kis.vh.stacks.types.StackArray;
+
 public class stack {
 
-	private int[] ITEMS = new int[12];	
+	private StackInterface stackI; // 7. przez utworzony obiekt będzie można przekazać dowolny
+									// obiekt implementujący interfejs StackInterface
+
+	public stack(StackInterface stackArray) {
```

Comment:
Czy jest to zawsze stackArray?

Reply:
Tutaj najwidoczniej zapomniałem zmienić nazwę. Do konstruktora jako argument może być przekazany obiekt dowolnej klasy implementujący interface StackInterface. Następnie ten obiekt jest przypisany do stackI, co również jest referencją do interface StackInterface. Można więc w tym przypadku użyć obiektu typu StackList.